skimage/morphology/extrema.py

```diff
@@ -207,152 +210,360 @@ def h_minima(image, h, selem=None):
+        connectivity is assumed, meaning `selem` is an array of 1's.
+    indices : bool, optional
+        If True, the output will be an array representing indices of local
+        maxima. If False, the output will be an array of 0's and 1's with the
+        same shape as `image`.
+    include_border : bool, optional
+        If true, plateaus that touch the image border are valid maxima.
 
-    Examples
+    Returns
+    -------
+    maxima : ndarray or tuple[ndarray]
+        If `indices` is false, an array with the same shape as `image` is
+        returned with 1's indicating the position of local maxima
+        (0 otherwise). If `indices` is true, a tuple of one-dimensional arrays
+        containing the coordinates (indices) of all found maxima.
+
+    See Also
     --------
-    >>> import numpy as np
-    >>> from skimage.morphology import extrema
+    skimage.morphology.local_minima
+    skimage.morphology.h_maxima
+    skimage.morphology.h_minima
 
-    We create an image (quadratic function with a maximum in the center and
-    4 additional constant maxima.
-    The heights of the maxima are: 1, 21, 41, 61, 81, 101
+    Notes
+    -----
+    This function operates on the following ideas:
 
-    >>> w = 10
-    >>> x, y = np.mgrid[0:w,0:w]
-    >>> f = 20 - 0.2*((x - w/2)**2 + (y-w/2)**2)
-    >>> f[2:4,2:4] = 40; f[2:4,7:9] = 60; f[7:9,2:4] = 80; f[7:9,7:9] = 100
-    >>> f = f.astype(np.int)
+    1. Make a first pass over the image's last dimension and flag candidates
+       for local maxima by comparing pixels in only one direction.
 
-    We can calculate all local maxima:
+    For each candidate:
 
-    >>> maxima = extrema.local_maxima(f)
+    2. Perform a flood-fill to find all connected pixels that have the same
+       gray value and are part of the plateau.
+    3. Consider the connected neighborhood of a plateau: if any no bordering
+       sample has a higher gray level, mark the plateau as a definite local
+       maximum.
 
-    The resulting image will contain all 6 local maxima.
+    Examples
+    --------
+    >>> from skimage.morphology import local_maxima
+    >>> image = np.zeros((4, 7), dtype=int)
+    >>> image[1:3, 1:3] = 1
+    >>> image[3, 0] = 1
+    >>> image[1:3, 4:6] = 2
+    >>> image[3, 6] = 3
+    >>> image
+    array([[0, 0, 0, 0, 0, 0, 0],
+           [0, 1, 1, 0, 2, 2, 0],
+           [0, 1, 1, 0, 2, 2, 0],
+           [1, 0, 0, 0, 0, 0, 3]])
+
+    Find local maxima by comparing to all neighboring pixels (maximal
+    connectivity):
+
+    >>> local_maxima(image)
+    array([[0, 0, 0, 0, 0, 0, 0],
+           [0, 1, 1, 0, 0, 0, 0],
+           [0, 1, 1, 0, 0, 0, 0],
+           [1, 0, 0, 0, 0, 0, 1]], dtype=uint8)
+    >>> local_maxima(image, indices=True)
+    (array([1, 1, 2, 2, 3, 3]), array([1, 2, 1, 2, 0, 6]))
+
+    Find local maxima without comparing to diagonal pixels (connectivity 1):
+
+    >>> local_maxima(image, selem=1)
```

Comment:
This is inconsistent with our existing APIs. `selem` should be a neighbourhood, like `footprint` for `ndimage` morphological functions; for integer connectivity, use `connectivity=1`. You should keep `selem` in for backwards compatibility, but otherwise please add `connectivity` to use integer input.